addons/purchase_stock/models/stock.py

```diff
@@ -186,28 +186,38 @@ def _compute_days_to_order(self):
         return res
 
     @api.depends('effective_route_id')
-    def _compute_show_supplier(self):
+    def _compute_show_partner(self):
         buy_route = []
         for res in self.env['stock.rule'].search_read([('action', '=', 'buy')], ['route_id']):
             buy_route.append(res['route_id'][0])
         for orderpoint in self:
-            orderpoint.show_supplier = orderpoint.effective_route_id.id in buy_route
+            orderpoint.show_partner = orderpoint.effective_route_id.id in buy_route
 
-    def _inverse_supplier_id(self):
-        for orderpoint in self:
-            if not orderpoint.route_id and orderpoint.supplier_id:
-                orderpoint.route_id = self.env['stock.rule'].search([('action', '=', 'buy')])[0].route_id
+    def _inverse_partner_id(self):
+        orderpoints = self.filtered(lambda orderpoint: not orderpoint.route_id and orderpoint.partner_id)
+        if orderpoints:
+            orderpoints.route_id = self.env['stock.rule'].search([('action', '=', 'buy')])[0].route_id
```

Comment:
```suggestion
    def _inverse_partner_id(self):
        for orderpoint in self:
            if not orderpoint.route_id and orderpoint.partner_id:
                orderpoint.route_id = self.env['stock.rule'].search([('action', '=', 'buy')])[0].route_id
```

Your idea is good to apply in batch, but never possible because we can't change the partner_id of multiple orderpoints at the same time, right?